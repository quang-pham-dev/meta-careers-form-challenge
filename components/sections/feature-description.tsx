interface FeatureDescriptionProps {
  title: string
  description: string
}

export function FeatureDescription({
  title,
  description,
}: FeatureDescriptionProps) {
  return (
    <div>
      <h3 className="text-xl font-semibold mb-2">{title}</h3>
      <p className="text-gray-600">{description}</p>
    </div>
  )
}
