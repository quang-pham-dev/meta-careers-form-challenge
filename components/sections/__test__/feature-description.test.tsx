import React from 'react'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

import { FeatureDescription } from '../feature-description'

describe('FeatureDescription', () => {
  const mockProps = {
    title: 'Test Title',
    description: 'Test Description',
  }

  it('matches snapshot', () => {
    const { asFragment } = render(<FeatureDescription {...mockProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the title correctly', () => {
    render(<FeatureDescription {...mockProps} />)
    const titleElement = screen.getByText(mockProps.title)
    expect(titleElement).toBeInTheDocument()
    expect(titleElement.tagName).toBe('H3')
    expect(titleElement).toHaveClass('text-xl font-semibold mb-2')
  })

  it('renders the description correctly', () => {
    render(<FeatureDescription {...mockProps} />)
    const descriptionElement = screen.getByText(mockProps.description)
    expect(descriptionElement).toBeInTheDocument()
    expect(descriptionElement.tagName).toBe('P')
    expect(descriptionElement).toHaveClass('text-gray-600')
  })
})
