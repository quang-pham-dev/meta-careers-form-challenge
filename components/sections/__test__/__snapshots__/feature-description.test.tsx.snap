// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeatureDescription > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="text-xl font-semibold mb-2"
    >
      Test Title
    </h3>
    <p
      class="text-gray-600"
    >
      Test Description
    </p>
  </div>
</DocumentFragment>
`;
